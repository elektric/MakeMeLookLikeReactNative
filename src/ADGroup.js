import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  Image,
  Switch,
} from 'react-native';

export default class ADGroup extends Component {
  constructor(props){
    super(props);
    this.state = {
            switchOn: true,
          };
  }
  render() {
    return (
        <View style={styles.splitViewOuter}>
          <View style={styles.splitViewInnerLeft}>
          <Text style={styles.appfuncStyle}>
            {this.props.details.Name}
          </Text>
        </View>
        <View style={styles.splitViewInnerRight}>
        <Switch style={styles.splitViewInnerRight} onValueChange={(value) => this.setState({switchOn: value})}
          value={this.state.switchOn} />
        </View>
      </View>
    );
  }

}

const styles = StyleSheet.create({

 appfuncStyle: {
   color: '#511',
   fontSize: 16,
   textAlign: 'center',
   marginBottom: 1,
   //backgroundColor: 'steelblue',
 },
 appfuncDescriptionStyle: {
   color: '#951',
   fontSize: 12,
   textAlign: 'center',
   marginBottom: 10,
   //backgroundColor: 'steelblue',
 },
 splitViewOuter: {
   flex: 1,
   flexDirection: 'row',
   justifyContent: 'space-around',
 },
 splitViewInnerLeft: {
   flex: 1,
   //justifyContent: 'center',
 },
 splitViewInnerRight: {
   //flex: 1,
   //justifyContent: 'center',
 },
});
