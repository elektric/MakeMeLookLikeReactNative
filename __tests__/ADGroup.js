import 'react-native';
import React from 'react';
import ADGroup from '../src/ADGroup';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('AD Group renders', () => {
  let adg = {'Name':'ADGroup7'};
  const tree = renderer.create(
    <ADGroup details={adg} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
