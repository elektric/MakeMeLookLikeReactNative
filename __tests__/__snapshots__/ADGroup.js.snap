// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AD Group renders 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#511",
          "fontSize": 16,
          "marginBottom": 1,
          "textAlign": "center",
        }
      }
    >
      ADGroup7
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <RCTSwitch
      disabled={false}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Object {},
        ]
      }
      value={true}
    />
  </View>
</View>
`;
